modules/services/kbfs.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.kbfs;
+
+in
+
+{
+  options = {
+    services.kbfs = {
+      enable = mkEnableOption "Keybase File System";
+
+      mountPoint = mkOption {
+        type = types.str;
+        default = "keybase";
+        description = ''
+          Mountpoint for the Keybase filesystem, relative to $HOME.
+        '';
+      };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = [
+          "-label kbfs"
+          "-mount-type normal"
+        ];
+        description = ''
+          Additional flags to pass to the Keybase filesystem on launch.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.kbfs = {
+      Unit = {
+        Description = "Keybase File System";
+        Requires = [ "keybase.service" ];
+        After = [ "keybase.service" ];
+      };
+  
+      Service = {
+        Environment = "PATH=/run/wrappers KEYBASE_SYSTEMD=1";
+        ExecStartPre = ''${pkgs.coreutils}/bin/mkdir -p "%h/${cfg.mountPoint}"'';
+        ExecStart = ''${pkgs.kbfs}/bin/kbfsfuse ${toString cfg.extraFlags} "%h/${cfg.mountPoint}"'';
+        ExecStopPost = ''/run/wrappers/bin/fusermount -u "%h/${cfg.mountPoint}"'';
```

Comment:
I suspect this will be a bit troublesome for non-NixOS systems but I don't see any particularly good way to avoid it. From what I can tell it is not possible to use, e.g., `udisksctl` to unmount a fuse mount point.

Reply:
I'm pretty sure kbfs itself is linux-only in nixpkgs, probably for the same reason.

Is there a way to specify the available platforms for a home-manager module?

Reply:
Unfortunately no. Maybe something could be added in the future. For now I think this is OK.